src/System.Runtime.Extensions/src/System/IO/TextReader.cs

```diff
@@ -100,6 +101,30 @@ public virtual int Read(char[] buffer, int index, int count)
             return n;
         }
 
+        // Reads a span of characters. This method will read up to
+        // count characters from this TextReader into the
+        // span of characters Returns the actual number of characters read.
+        //
+        public virtual int Read(Span<char> destination)
+        {
+            char[] buffer = ArrayPool<char>.Shared.Rent(destination.Length);
+
+            try
+            {
+                int numRead = Read(buffer, 0, destination.Length);
+                if ((uint)numRead > destination.Length)
+                {
+                    throw new IOException(SR.IO_StreamTooLong);
```

Comment:
There's not necessarily a Stream involved here, so the error name and message is a bit misleading.  I suggest something instead like "The read operation returned an invalid length."

Reply:
Done...